cmd/kubectl-gadget/top/block-io.go

```diff
@@ -30,84 +31,137 @@ import (
 	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/top/block-io/types"
 )
 
-var blockIONodeStats map[string][]types.Stats
+type BlockIOFlags struct {
+	CommonTopFlags
 
-// flags
-var blockIOSortBy types.SortBy
+	ParsedSortBy types.SortBy
+}
 
-var blockIOCmd = &cobra.Command{
-	Use:   fmt.Sprintf("block-io [interval=%d]", types.IntervalDefault),
-	Short: "Periodically report block device I/O activity",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		var err error
+type BlockIOParser struct {
+	commonutils.BaseParser[types.Stats]
+	sync.Mutex
 
-		blockIONodeStats = make(map[string][]types.Stats)
+	flags     *BlockIOFlags
+	nodeStats map[string][]types.Stats
+}
 
-		if len(args) == 1 {
-			outputInterval, err = strconv.Atoi(args[0])
-			if err != nil {
-				return commonutils.WrapInErrInvalidArg("<interval>",
-					fmt.Errorf("%q is not a valid value", args[0]))
+func newBlockIOCmd() *cobra.Command {
+	var flags BlockIOFlags
+
+	commonFlags := &utils.CommonFlags{
+		OutputConfig: commonutils.OutputConfig{
+			// The columns that will be used in case the user does not specify
+			// which specific columns they want to print.
+			CustomColumns: []string{
+				"node",
+				"namespace",
+				"pod",
+				"container",
+				"pid",
+				"comm",
+				"r/w",
+				"major",
+				"minor",
+				"bytes",
+				"time",
+				"ios",
+			},
+		},
+	}
+
+	columnsWidth := map[string]int{
+		"node":      -16,
+		"namespace": -16,
+		"pod":       -30,
+		"container": -16,
+		"pid":       -7,
+		"comm":      -16,
+		"r/w":       -3,
+		"major":     -6,
+		"minor":     -6,
+		"bytes":     -7,
+		"time":      -8,
+		"ios":       -8,
+	}
```

Comment:
Those information would be really good to have available more generically (through an interface for example) - but that's probably out of scope here.

Reply:
Absolutely agree! Let's explore some ideas for a future PR.